src/app/reporting/Engine.cpp

```diff
@@ -47,6 +47,21 @@ void Engine::Shutdown()
     mGlobalDirtySet.ReleaseAll();
 }
 
+bool Engine::IsClusterDataVersionMatch(ClusterInfo * aDataVersionFilterList, const ConcreteReadAttributePath & aPath)
+{
+    for (auto filter = aDataVersionFilterList; filter != nullptr; filter = filter->mpNext)
+    {
+        if (aPath.mEndpointId == filter->mEndpointId && aPath.mClusterId == filter->mClusterId && filter->mDataVersion.HasValue())
+        {
+            if (IsClusterDataVersionEqual(filter->mEndpointId, filter->mClusterId, filter->mDataVersion.Value()))
+            {
+                return true;
```

Comment:
This is still not implementing the spec-defined behavior, right?  That behavior is:
> If the DataVersionFilters field does not indicate a DataVersionFilterIB entry with a Path field that matches the cluster path, or the DataVersion field of a matching DataVersionFilterIB does not match the data version of the cluster instance in the path, then an AttributeDataIB SHALL be generated with the attribute data for the path and cluster data version with the following filters:

That's not what this function implements.

Please fix and add a test.

Reply:
fixed, and test added, thanks